src/server.c

```diff
@@ -1648,8 +1648,15 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         return;
     }
 
-    /* Handle precise timeouts of blocked clients. */
-    handleBlockedClientsTimeout();
+    if (moduleCount())
+        moduleFireServerEvent(REDISMODULE_EVENT_EVENTLOOP,
+                              REDISMODULE_SUBEVENT_EVENTLOOP_BEFORE_SLEEP,
+                              NULL);
+
+    /* Handle blocked clients.
+     * must be done before flushAppendOnlyFile, in case of appendfsync=always,
+     * since the unblocked clients may write data. */
+    blockedBeforeSleep();
```

Comment:
isn't it better to do these after `connTypeProcessPendingData()` and also after `handleClientsWithPendingReadsUsingThreads()`?

i.e. these can also trigger some commands to be processed.
or maybe it's a chicken and an egg thing? i don't think so...

Reply:
i'd like to challenge the difference between what i suggested and what you changed.
you a few more things that i didn't mean for:
1. aeSetDontWait to happen before processing blocked clients. i suppose that doesn't make any difference, since in either case we're done processing sockets, so the sampling of connTypeHasPendingData is probably the same.
2. sample cron_start_time_before_aof to be done before processing blocked clients. arguably that's wrong since these things are not "cron", i.e. unlike active expiry.
3. move moduleFireServerEvent to after these (handleClientsWithPendingReadsUsingThreads, connTypeProcessPendingData, and cron_start_time_before_aof), i'm not sure it makes any difference with regards to the first two, regarding the 3rd, maybe it's even more "correct" (to count that as cron)

WDYT? 

i'm not certain

Reply:
I'll move blockedBeforeSleep to be above cron_start_time_before_aof sampling